### PR TITLE
[ALCA] Fix copy-constructor warnings reported in DEVEL IBs

### PR DESCRIPTION
**Pull request by aandvalenzuela**
Hello,

This PR follows up on the discussion started at https://github.com/cms-sw/cmssw/pull/43408.
I am defining the missing assignment operators to fulfill the rule of three (see https://github.com/cms-sw/cmssw/pull/43408#issuecomment-1829333134) to avoid missing initializations (see https://github.com/cms-sw/cmssw/pull/43408#discussion_r1407565529)

Thanks!

### CODE REVIEW

File: CalibFormats/SiStripObjects/interface/SiStripHashedDetId.h [hunk 1]
```diff
@@ -29,6 +29,9 @@ class SiStripHashedDetId {
   /** Copy constructor. */
   SiStripHashedDetId(const SiStripHashedDetId &);
 
+  /** Assignment operator. */
+  const SiStripHashedDetId &operator=(const SiStripHashedDetId &) { return *this; }
```
Comment (iarspider) [inline CalibFormats/SiStripObjects/interface/SiStripHashedDetId.h:33]: @aandvalenzuela are you sure this is correct?

File: CalibCalorimetry/EcalPedestalOffsets/interface/TSinglePedEntry.h [hunk 2]
```diff
@@ -15,12 +15,14 @@ class TSinglePedEntry {
   //! ctor
   TSinglePedEntry();
   //! copy ctor
-  TSinglePedEntry(const TSinglePedEntry &orig);
+  TSinglePedEntry(const TSinglePedEntry& orig);
+  //! assignment op
+  const TSinglePedEntry& operator=(const TSinglePedEntry& orig) { return *this; }
```
Comment (iarspider) [inline CalibCalorimetry/EcalPedestalOffsets/interface/TSinglePedEntry.h:20]: @aandvalenzuela are you sure this is correct?

File: CalibFormats/SiStripObjects/interface/SiStripHashedDetId.h (hunk 1 above)
Comment (aandvalenzuela) [reply CalibFormats/SiStripObjects/interface/SiStripHashedDetId.h:33]: No, I am not. This is what the code format reported (see https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-43423/37938/code-format.patch). I was actually taking this issue offline...

Comment (iarspider) [reply CalibFormats/SiStripObjects/interface/SiStripHashedDetId.h:33]: I mean - are you sure you have implemented `operator=` correctly?

Comment (aandvalenzuela) [reply CalibFormats/SiStripObjects/interface/SiStripHashedDetId.h:33]: I think that is the default implementation, but I am not familiar with it. Do you have any suggestions?

Comment (aandvalenzuela) [reply CalibFormats/SiStripObjects/interface/SiStripHashedDetId.h:33]: Does something like this look better?
```
TSinglePedEntry& operator=(const TSinglePedEntry& orig) {
    m_pedestalSqSum = orig.m_pedestalSqSum;
    m_pedestalSum = orig.m_pedestalSum;
    m_entries = orig.m_entries;
    return *this; 
 }
```

**Comment by aandvalenzuela** [comment]
please test

**Comment by aandvalenzuela** [comment]
please test for CMSSW_14_0_DEVEL_X

**Comment by perrotta** [comment]
+alca
- Default assignment operators added

**Pull request merged**